### PR TITLE
type_with_subtypest now has constructors that take the subtypes

### PR DESCRIPTION
**Pull request by kroening**
- [ ] Each commit message has a non-empty body, explaining why the change was made.
- [X] My contribution is formatted in line with CODING_STANDARD.md.
- [ ] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [ ] My commit message includes data points confirming performance improvements (if claimed).
- [X] My PR is restricted to a single feature or bugfix.
- [X] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: 3d53d3e07a8ef8d796f6985efdf85abe67206d7b).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/87880016

**Review by smowton — APPROVED** [review]

File: src/util/type.h [hunk 1]
```diff
@@ -114,11 +114,17 @@ inline type_with_subtypet &to_type_with_subtype(typet &type)
...
   type_with_subtypest() { }
 
   explicit type_with_subtypest(const irep_idt &_id):typet(_id) { }
 
-  typedef std::vector<typet> subtypest;
+  type_with_subtypest(const irep_idt &_id, const subtypest &_subtypes):
+    typet(_id) { subtypes()=_subtypes; }
+
+  type_with_subtypest(const irep_idt &_id, subtypest &&_subtypes):
+    typet(_id) { subtypes()=std::move(_subtypes); }
```
Comment (tautschnig) [inline src/util/type.h]: Nit picking: would it be possible to fix the formatting?

Comment (kroening) [reply src/util/type.h]: Done

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: e3bbd580c33002793ab3437c730071c1ec7119f0).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/87890142

**Pull request merged**